generator/sequence_transform_data.py

```diff
@@ -187,109 +187,97 @@ def make_sequence_trie(
     output_func_symbol_map: Dict[str, int]
 ) -> Dict[str, tuple[str, dict]]:
     """Makes a trie from the sequences, writing in reverse."""
```

Comment:
`make_sequence_trie` now combines the functionality of the old version of `make_sequence_trie` and `complete_sequeuce_trie`.

Chained rules always chain exclusively off of rules with sequeuces that are prefixes of the chained rules prefix. That is, `_k@d -> _knowledge` chains of of `_k@ -> _know` and cannot chain off of `k@ -> know`. In this PR, the system doesn't work as you might expect for a rule (call it RULE) if there is any other rule that has a seq which is a substring of the RULE.sequence without also being either a prefix of RULE.sequence or being a suffix of sequence that is a prefix of RULE.sequence. That is confusing, so I will use a real life example:
Consider the following ruleset:
```
w* -> way
_w*r -> write
```
The second rule can't chain off the first, so it ignores it, and the action it `{backspaces = 1, completion = rite}`, and typing `_w*t` produces `warite`, because the `w*` rule messes things up.
The fix is to add a third rule `_w* -> _way`. Now the `_w*r -> write` rule chains off the new rule and has an action of `{backspaces = 2, completion = rite}`. And if we type `_w*r`, the first rule never triggers (because it is a suffix of the longer new rule that matches instead), and the `_w*r` rule successfully triggers off the sub-match of the new rule.

We should at least warn that there are conflicting rules here, and suggest the fix. But I think we can go a step further and simply automatically generate these missing intermediate rules. If a user has the `w* -> way` rule, then adding the `_w* -> _way` rule changes no behavior, except for allowing the chained rule to work correctly. More-over, adding this intermediate rule to the trie actually *decreases* its size.

My plan is to add the auto-rule generation and test it on @QKekos and @kamih 's rulesets to see if I'm overlooking something. The auto-generated rules will be included in the comment section of the generated `sequence_transform_data.h` with a note that it was autogenerated.